### PR TITLE
fix: support IE9

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Recovers support for IE9 and maintains supporting custom variables.

Support for `!important` still does not work but I wonder if this is even needed, it does however if people use it disable the full propagation of the prop.

`style={{ background: 'blue !important' }}` --> evaluates to `style={{}}`. This is pretty akward for devs relying on it. If we would want to support !important we would have to use `setProperty` and neglect the bug with IE9. setProperty can set !important with its thirth argument.

I do feel like that maybe we support the majority of IE9 and then we're fine but this could also open up a rabbit hole against future features. (this is when we'd officially declare IE9 support)

ADDS: ~~12~~ ~~10B~~ 9B

### CODE REVIEW

File: src/diff/props.js [hunk 1]
```diff
@@ -59,7 +59,12 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 			for (let i in value) {
 				v = value[i];
 				if (oldValue==null || v!==oldValue[i]) {
-					s.setProperty(i.replace(CAMEL_REG, '-'), typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v);
+					if (i.indexOf('--')===0) {
```
Comment (marvinhagemeister) [inline src/diff/props.js]: Just using `!i.indexOf('--')` saves 2 bytes :tada:

**Comment by marvinhagemeister** [comment]
I'd be happy to add it just for supporting `!important` alone :tada: Thanks for diving in to this :100:

**Comment by JoviDeCroock** [comment]
I don't think we can support !import even with this, I still have to explore that part.

**Comment by marvinhagemeister** [comment]
Oh ok, if that's the case I'd vote against adding IE9 specific code.

**Comment by JoviDeCroock** [comment]
So rather have it catch and replace important to the thirth argument?

File: src/diff/props.js [hunk 2]
```diff
@@ -59,7 +59,12 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 			for (let i in value) {
 				v = value[i];
 				if (oldValue==null || v!==oldValue[i]) {
-					s.setProperty(i.replace(CAMEL_REG, '-'), typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v);
+					if (!i.indexOf('--')) {
```
Comment (developit) [inline src/diff/props.js]: I believe this would be smaller and faster:

```suggestion
					if (i[0]=='-') {
```

**Comment by cloudkite** [comment]
Hey 👋Just chiming in as I feel like my comment in https://github.com/developit/preact/issues/1482 may have been misleading. I was under the impression that React supported !important in inline styles, was curious so I tried it today in a React project and it does not work.

**Comment by JoviDeCroock** [comment]
Closing this since the decision made involves:
- style[key] = value is slow, this punishes all browsers except IE9
- !important can be supported through style="color: red !important"

**Comment by developit** [comment]
Bench from the above:

**Pull request closed without merging**